gputop/remote/ajax/metrics.html

```diff
@@ -78,7 +78,12 @@ <h2 id="page-header-metrics">Metrics panel loading</h2>
         var symbol_name = $counter.attr("symbol_name");
         var description = $counter.attr("description") + " (" + $counter.attr("mdapi_usage_flags") + ")";
         var flags = $counter.attr("mdapi_usage_flags").split(" ");
-        set_array[set_array.length - 1].counters.push(new count(name, symbol_name, flags));
+        for (var key in set_array)
+        {
+            if (set_array.hasOwnProperty(key))
+                selected_index = key;
+        }
+        set_array[selected_index].counters.push(new count(name, symbol_name, flags));
```

Comment:
My JavaScript foo is quite limited, but I'm not quite sure about this loop...

I think the `key in object` syntax is almost synonymous with `object.hasOwnProperty(key)` except that the `in` operator doesn't only check the very specific given `object` but will also walk through the prototype chain of the object to see if any of those has that property. `hasOwnProperty()` doesn't fall back to walking through the prototype chain looking for the property.

I'm not totally sure so far, but I assume we aren't touching the prototype chain of the set_array so I think in this case the two are effectively equivalent tests so we can assume the `if (set_array.hasOwnProperty(key))` check will always be true.

I think the result of this loop is to initialize `selected_index` to the last iterated `key` found in `set_array`, but I'm not sure there are any guarantees about which key that is, since I don't know that there's an inherent order to object properties afaik.